src/main.rs

```diff
@@ -60,3 +76,41 @@ fn main() -> anyhow::Result<()> {
     fs::write("gitice.lock", toml::to_string(&repos)?).expect("could not write to lockfile!");
     Ok(())
 }
+
+fn thaw_repos(dir: String) -> anyhow::Result<()> {
+    let lockfile = fs::read_to_string("gitice.lock").expect("unable to read lockfile!");
+    let repos: HashMap<String, PersistableRepo> = toml::from_str(&lockfile)?;
+
+    let mut callbacks = RemoteCallbacks::new();
+    callbacks.credentials(|_url, username_from_url, _allowed_types| {
+        Cred::ssh_key(
+            username_from_url.unwrap(),
+            None,
+            Path::new(&format!(
+                "{}/.ssh/id_rsa",
```

Comment:
I believe cargo shells out to the Git binary for cloning because it's simpler than implementing all the fallback cases for authentication so maybe we should too.

Reply:
Hm, that sounds better